pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -206,29 +254,41 @@ func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node, record
 		return err
 	}
 
-	nodeInternalIPs, err := dnshelpers.GetInternalIPAddressesForNodeName(node)
+	// create the certificates and update them in the API
+	pCert, pKey, err := tlshelpers.CreatePeerCertKey(etcdCASecret.Data["tls.crt"], etcdCASecret.Data["tls.key"], nodeInternalIPs)
 	if err != nil {
 		return err
 	}
 
-	// create the certificates and update them in the API
-	pCert, pKey, err := tlshelpers.CreatePeerCertKey(etcdCASecret.Data["tls.crt"], etcdCASecret.Data["tls.key"], nodeInternalIPs)
-	err = c.createSecret(etcdPeerClientCertName, pCert, pKey, recorder)
-	if err != nil {
+	if err := c.createSecret(etcdPeerClientCertName, pCert, pKey, recorder); err != nil {
 		return err
 	}
 	sCert, sKey, err := tlshelpers.CreateServerCertKey(etcdCASecret.Data["tls.crt"], etcdCASecret.Data["tls.key"], nodeInternalIPs)
-	err = c.createSecret(etcdServingCertName, sCert, sKey, recorder)
 	if err != nil {
 		return err
 	}
+
+	if err := c.createSecret(etcdServingCertName, sCert, sKey, recorder); err != nil {
+		return err
+	}
+
 	metricCert, metricKey, err := tlshelpers.CreateMetricCertKey(etcdMetricCASecret.Data["tls.crt"], etcdMetricCASecret.Data["tls.key"], nodeInternalIPs)
-	err = c.createSecret(metricsServingCertName, metricCert, metricKey, recorder)
 	if err != nil {
 		return err
 	}
 
-	return nil
+	if err := c.createSecret(metricsServingCertName, metricCert, metricKey, recorder); err != nil {
+		return err
+	}
+
+	_, _, updateErr = v1helpers.UpdateStatus(c.operatorClient,
+		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "EtcdCertSignerControllerDegraded",
+			Status: operatorv1.ConditionFalse,
+			Reason: "AsExpected",
+		}))
+
+	return updateErr
```

Comment:
same as above status is set in sync based on the result of the sync function syncAllMasters